python_modules/dagster/dagster/_serdes/serdes.py

```diff
@@ -1193,6 +1200,10 @@ def _unpack_value(
 def _check_serdes_tuple_class_invariants(
     klass: Type[NamedTuple], is_pickleable: bool = True
 ) -> None:
+    # can skip validation on programmatically generated new
+    if klass.__new__.__name__ == "__checked_new__":
+        return
+
```

Comment:
<!--__GRAPHITE_HTML_TAG_START__--><i class='graphite__hidden'>[Re: lines 646 to 1206]<br/><br/></i>

<!--__GRAPHITE_HTML_TAG_END__-->helper function `has_autogenned_checked_new` or something would be nice, co-located with a documented constant.<!--__GRAPHITE_HTML_TAG_START__--><span class='graphite__hidden'><br/><br/>See this comment inline on <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/22043?utm_source=unchanged-line-comment">Graphite</a>.</span>

<!--__GRAPHITE_HTML_TAG_END__-->